tensorizer/stream_io.py

```diff
@@ -142,6 +142,7 @@ def __init__(
         begin: Optional[int] = None,
         end: Optional[int] = None,
         headers: Dict[str, Any] = None,
+        buffer_size: int = 2**21,  # `2mb` buffer on the python IO object
```

Comment:
We can make this a keyword-only argument to make it easier to change later
```suggestion
        *,
        buffer_size: int = 2 << 20,  # 2 MiB buffer on the Python IO object
```